pom.xml

```diff
@@ -383,6 +378,32 @@
         <artifactId>snakeyaml</artifactId>
         <version>${snakeyaml.version}</version>
       </dependency>
+      <dependency>
+        <groupId>org.apache.hadoop</groupId>
+        <artifactId>${hadoop-client-api.artifact}</artifactId>
+        <version>${hadoop.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.hadoop</groupId>
+        <artifactId>${hadoop-client-runtime.artifact}</artifactId>
+        <version>${hadoop.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.hadoop</groupId>
+        <artifactId>hadoop-mapreduce-client-app</artifactId>
```

Comment:
why should we introduce this dep?

Reply:
Hadoop 3 `MRAppMaster` is in this dependency.

Reply:
this actually pulls hadoop-common deps, which is unexpected.

Reply:
BTW, is it already under MR job classpath? if then, it could be marked as provided